_posts/2017-11-07-amp-roadshow.md

```diff
@@ -0,0 +1,114 @@
+<br/><em>Overview</em>
+</div>
+
+Next, Raghu told us about some of the technology inside AMP. When he said that AMP is a [web components] library I was [all ear][tweet-1], since that is the thing I am currently learning and evaluating. The later discussion with Paul also made me understand that AMP's approach to web components has a focus on fully controlling the web site's beahvior and not on providing (UI) components in the first place. That [took me a little][tweet-2] to understand. But in the AMP context it makes sense.
+
+AMP re-implements many DOM elements we know from HTML, that interact with "slow" external dependencies, such as loading images (and rendering them). AMP focuses on optimizing the user experience, which is rendering speed to name the most important one. For example in AMP you use `<amp-img>` instead of `<img>`. The custom elements `<amp-youtube>`, `<amp-video>` and `<amp-twitter>` make it very obvious what AMP wants to control: the loading pipeline of anything that might make the rendering slow. There are various extensions though, so there might be many even more customized elements that can be used with AMP.
+
+While the reasoning behind it and the techniques used make a lot of sense, AMP also limits down the possibility of what one can use. Because if you want your site to be a valid AMP site you need to run it against the [AMP validator][amp-validator], which checks the site to fulfill certain criteria, for example that all CSS must be inlined and must not be exceed 50kB.
+
+[web components]: https://www.webcomponents.org/
+[tweet-1]: https://twitter.com/wolframkriesing/status/927825238766882818
+[tweet-2]: https://twitter.com/wolframkriesing/status/927835789404319744
+[amp-validator]: https://validator.ampproject.org/
+
+## AMP in Production
+
+<div style="float: left; padding: 1rem;">
+<img src="{{site.baseurl}}/img/posts/2017-11-amp-roadshow/3-in-prod.jpg" alt="amp in production" width="100" class="sizeup-onhover-image scale3 origin-left-top" />
+<br/><em>AMP in production</em>
+</div>
+
+After the basics, the "why" and "how" of AMP, [Rudy][rudy] (???????? was it Rudy?????????) went to show us about how to get AMP live. Three basic steps he talked about 1) Validate 2) Distribute and 3) Measure. After an AMP page ran through the validator one wants to take it live. When this comes up the measuring part is an important one. Some sites need only simple page view tracking, others even want A/B tests. AMP tries to serve them all. There is a `<amp-analytics>` custom element that unifies the analytics work under one hood. It basically serves as a proxy on the client, I assume it has a fixed set of tracking data that it collects and does then spread it to the configured backends (analytics provider). This allows AMP to do the tracking and not every tracker to bring its own code, polute your site and potentially crash it.
```

Comment:
`one wants to take it live` -> `it can be put into production`
`When this comes up` -> `Afterwards`